game/src/select_cards.rs

```diff
@@ -49,13 +52,343 @@ fn get_crib_cards(hand: &[Card], held_cards: &[Card]) -> Vec<Card> {
     }
     send_to_crib
 }
+/**
+ *  called during counting phase
+ *  this api looks at the cards that have already been played and then the cards that could be played and tries to pick the right one
+ *
+ *  this is where "strategy" is implemented as selecting what cards to give to the crib is straightforward probability whereas counting
+ *  cards is largely a function of anticipating what the other player is going to play -- e.g. if you have a 2 and a 3, you might play the
+ *  2 hoping that the other player has only cards with value = 10, so that you can play your 3 and get two points for the 15.  But the other
+ *  player knows this is a normal thing to do, so by playing a 2 you imply you also have a 3.  So maybe the right card to play is a 4 so that the
+ *  apponent plays the 3 to get the run (2, 4, 3) and 3 points - and then bam! -- you play an Ace to get 4 points.  and so on.
+ *
+ *  my strategy for this is to implement it the way I play the game.  by its nature it is a bunch of "if this then that" calls, so this function
+ *  tends to be very long and very experimental.  I have added logic in the past and then tested it by playing millions (literally) of games of
+ *  one algorithm against the other to see if it really works (it takes about 20 seconds to run a million games on a beefy PC...)
+ *
+ *
+ */
+pub fn get_next_counted_card(
+    counted_cards: Vec<Card>,
+    mut cards_left: Vec<Card>, // needs to be mut because we .sort() it
+) -> Result<Card, Box<dyn Error>> {
```

Comment:
Better semantics might be Option<Card> and the semantics of the call are either I have a card to play or I don't. Some(...) is a play and None is a Go. I wouldn't treat Go as an Error, given that it's a thing that we expect to happen.